src/StageDisturb.jl

```diff
@@ -0,0 +1,75 @@
+mutable struct StageDisturb
+    data::TSP
+    solution::Solution
+    neighbourSearch::Vector{Any}
+    currentIteration::Int
+    stageThresholds::Vector{Int}
+    disturbFunctions::Vector{Function}
+    gap::Int
+ end
+
+ function iteratedLocalSearch(stageDisturb::StageDisturb)
+    logfile = open("log.txt", "a")
+
+    sol = stageDisturb.solution
+    
+    @info "Initial solution" sol
+
+    bestSol = deepcopy(stageDisturb.solution)
+    maxIter = stageDisturb.stageThresholds[end]
+    disturb = ShuffleSublist(instance, sol, stageDisturb.gap)
+
+    while stageDisturb.currentIteration < maxIter
+        logAndPrint(logfile, @sprintf("Iteration %d", sd.currentIteration))
+
+        room_to_improve = true
+        while room_to_improve
+            room_to_improve = false
+            shuffle!(stageDisturb.neighbourSearch)
+            for ns in stageDisturb.neighbourSearch
+                if bestImprovement!(ns)
+                    room_to_improve = true
+                    updateCost!(ns.solution, ns.data.weights)
+                    break
+                else 
+                    updateCost!(ns.solution, ns.data.weights)
+                end
+            end
+        end
+
+        currSol = stageDisturb.neighbourSearch[1].solution
```

Comment:
why do you always set currSol as the solution from the first neighborhood? if neighbourSearch is shuffled somewhere, then I agree with this, but I didn't see it

Reply:
All neighbourSearchs point to the same array, so it is indifferent which one I use to update currSol

Reply:
@lucasximenes I checked the code again and you are right, I was deepcopying to every NS when I meant to create a copy and pass the reference. I've pushed the new changes.